[BUG] Systemd integTest did not check for opensearch:adm on deb packages

### Describe the bug

On deb we see this error now:
```
Suite: Test class org.opensearch.systemdinteg.SystemdIntegTests

  2> Mar 16, 2025 6:11:26 AM org.apache.lucene.internal.vectorization.VectorizationProvider lookup

  2> WARNING: C2 compiler is disabled; Java vector incubator API can't be enabled

  2> org.junit.ComparisonFailure: Path should be owned by opensearch:opensearch expected:<opensearch:[opensearch]> but was:<opensearch:[adm]>

        at __randomizedtesting.SeedInfo.seed([3652A40F60668475:F813FFC392254131]:0)

        at org.junit.Assert.assertEquals(Assert.java:117)

        at org.opensearch.systemdinteg.SystemdIntegTests.testReadWritePaths(SystemdIntegTests.java:119)

  2> NOTE: reproduce with: gradlew test --tests SystemdIntegTests.testReadWritePaths -Dtests.seed=3652A40F60668475 -Dtests.locale=mer-KE -Dtests.timezone=UCT -Dtests.asserts=true -Dtests.file.encoding=UTF-8

  2> java.lang.AssertionError: Swap system call should be blocked

        at __randomizedtesting.SeedInfo.seed([3652A40F60668475:6A11414F2819C8A0]:0)

        at org.junit.Assert.fail(Assert.java:89)

        at org.junit.Assert.assertTrue(Assert.java:42)

        at org.opensearch.systemdinteg.SystemdIntegTests.testSystemCallFilter(SystemdIntegTests.java:147)

  2> NOTE: reproduce with: gradlew test --tests SystemdIntegTests.testSystemCallFilter -Dtests.seed=3652A40F60668475 -Dtests.locale=mer-KE -Dtests.timezone=UCT -Dtests.asserts=true -Dtests.file.encoding=UTF-8

  2> NOTE: test params are: codec=DummyCompressingStoredFieldsData(storedFieldsFormat=Lucene90CompressingStoredFieldsFormat(compressionMode=DUMMY, chunkSize=17090, maxDocsPerChunk=1, blockShift=21), termVectorsFormat=Lucene90CompressingTermVectorsFormat(compressionMode=DUMMY, chunkSize=17090, maxDocsPerChunk=1, blockSize=21)), sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=mer-KE, timezone=UCT

  2> NOTE: Linux 6.1.109-118.189.amzn2023.x86_64 amd64/Eclipse Adoptium 21.0.6 (64-bit)/cpus=16,threads=1,free=502776424,total=538968064

  2> NOTE: All tests run in this JVM: [SystemdIntegTests]
```

On @RajatGupta02  initial PR it is testing for `opensearch:opensearch` ownership which is not what we defined for deb which is `opensearch:adm`:
https://github.com/opensearch-project/OpenSearch/pull/17410/files#diff-de1c6d21b56c4b8dc96bbb92655860500a588d64019cb17fbe83c6eafd0c1d3fR119

As can be seen in this section:
https://github.com/opensearch-project/opensearch-build/blob/main/scripts/pkg/build_templates/current/opensearch/deb/debian/postinst#L44

Which we did not enforce on rpm due to a community contributor PR.

Let me know if we can relax the check from `opensearch:opensearch` only to either `opensearch:opensearch` or `opensearch:adm`.

Thanks.

### Related component

Other

### To Reproduce

See description.

### Expected behavior

See description

### Additional Details

See description

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Change priority for scheduling reroute during timeout([#16445](https://github.com/opensearch-project/OpenSearch/pull/16445))
 - Renaming the node role search to warm ([#17573](https://github.com/opensearch-project/OpenSearch/pull/17573))
 - Introduce a new search node role to hold search only shards ([#17620](https://github.com/opensearch-project/OpenSearch/pull/17620))
+- Fix systemd integTest on deb regarding path ownership check ([#17641](https://github.com/opensearch-project/OpenSearch/pull/17641)) 
 
 ### Dependencies
 - Bump `ch.qos.logback:logback-core` from 1.5.16 to 1.5.17 ([#17609](https://github.com/opensearch-project/OpenSearch/pull/17609))

--- a/qa/systemd-test/src/test/java/org/opensearch/systemdinteg/SystemdIntegTests.java
+++ b/qa/systemd-test/src/test/java/org/opensearch/systemdinteg/SystemdIntegTests.java
@@ -116,7 +116,8 @@ public class SystemdIntegTests extends LuceneTestCase {
             assertTrue("Path should exist: " + path, checkPathExists(path));
             assertTrue("Path should be readable: " + path, checkPathReadable(path));
             assertTrue("Path should be writable: " + path, checkPathWritable(path));
-            assertEquals("Path should be owned by opensearch:opensearch", "opensearch:opensearch", getPathOwnership(path));
+            String ownership = getPathOwnership(path);
+            assertTrue("Path should be owned by opensearch:opensearch or opensearch:adm", ownership.equals("opensearch:opensearch") || ownership.equals("opensearch:adm"));
         }
     }
 

